### ISSUE TITLE
Stache views which mutate tables may crash Internet Explorer 11

### ISSUE THREAD

**Issue by rjgotten** [canjs/can-stache#94]
Internet Explorer 11 can not cope with DOM mutations to tables which do not have `<tbody>` elements and which are (directly or indirectly with `subtree:true`) being observed by a `MutationObserver`.

They cause a hard crash of the entire browser with a fault code similar to `Unhandled exception at 0x05D424C1 (mshtml.dll) in iexplore.exe: 0xC0000005: Access violation reading location 0x00000008.`

Stache should probably take care to ensure that `<tbody>` elements are always present, or should not use `MutationObserver` for IE 11. However, given that any other code operating on a page might still be registering mutation observers, the first (ensuring that `<tbody>` is always present) is probably a better idea.

Likewise, IE 11 has additional problems with crashes when using `insertAdjacentHTML` on a `<tbody>` element, which should probably be guarded against as well.

See e.g. https://github.com/meteor/meteor/issues/6081 for Meteor, which ran into problems with this curiosity as well.

**Comment by justinbmeyer** [comment]
Interesting.  Have you seen this happen in stache?

[This gist](https://gist.github.com/josh/7597309) in that meteor issue shows it happening when `insertAdjacentHTML` is called. We don't use `insertAdacentHTML`, does it happen with any `appendChild` etc?

**Comment by rjgotten** [comment]
> @justinbmeyer 
> Interesting. Have you seen this happen in stache?

Yes. Multiple times. Both in stache and in EJS, actually. I had to rewrite a core piece of functionality used in multiple of my employer's projects to work around this issue, before a collleague of mine realized by freak chance that the culprit was to be found inside a `<tbody>`-less table.

That bit of information in turn allowed us to drastically narrow our search for a root cause, which lead us to discover the above bug report for Meteor that signals the combination of (lack of) `<tbody>` and use of `MutationObserver` to be the cause of our woes.

Also, from my own testing I found that this doesn't just trigger on `<table>` elements and friends, but infact may trigger on anything that uses `display:table`.


To recap and clarify: there are **two** problems here.
IE 11 may crash

  1. when any type of DOM mutation occurs inside an element that is

    * layed out as a table; and
    * lacks an explicit child element that is layed out as a table body; and
    * is directly or indirectly observed by a `MutationObserver`

    ; or

  2. when the `insertAdjacentHTML` method is used on any element that is:

     * layed out as a table body; and
     * is directly or indirectly observed by a `MutationObserver`

**Comment by bmomberger-bitovi** [comment]
So retesting against IE11 (build 11.0.9600.17416) on Windows 8.1, the linked gist doesn't cause IE to crash (and I can't seem to make an observed table that causes the crash).  Closing for now, but if you still see the bug please provide your build number for IE (and Windows version) and reopen.
